File: java/src/main/java/dominio/estados/EstadoConcretoA.java
package dominio.estados;

import dominio.Contexto;

public class EstadoConcretoA implements EstadoContexto{

    public void actuar(Contexto contexto) {
        //TODO
        if(cumpleCondicionParaCambiarDeEstado(contexto)){
            contexto.cambiarEstado(new EstadoConcretoB());
        }
    }

    private boolean cumpleCondicionParaCambiarDeEstado(Contexto contexto){
        //TODO
        return true;
    }
}

File: php/src/app/dominio/estados/EstadoConcretoA.php
<?php
namespace dominio\estados;

use dominio\Contexto;

class EstadoConcretoA implements EstadoContexto {

    public function actuar(Contexto $contexto){
        // TODO: Implement actuar() method.
        if($this->cumpleCondicionParaCambiarDeEstado($contexto)){
            $contexto->cambiarEstado(new EstadoConcretoB());
        }
    }

    private function cumpleCondicionParaCambiarDeEstado(Contexto $contexto) : boolean {
        // TODO
        return true;
    }
}